src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -796,6 +796,27 @@ public override void VisitSwitchExpression(SwitchExpressionSyntax node)
             }
         }
 
+        public override void VisitBinaryPattern(BinaryPatternSyntax node)
+        {
+            // Users (such as ourselves) can have many, many nested binary patterns. To avoid crashing, do left recursion manually.
+            PatternSyntax currentPattern = node;
+
+            var rightPatternStack = ArrayBuilder<PatternSyntax>.GetInstance();
```

Comment:
>var rightPatternStack = ArrayBuilder<PatternSyntax>.GetInstance(); [](http://example.com/codeflow?start=12&length=66)

I think the order of visiting arms is not important for the purpose of creating binders. Therefore, we don't really need to use manual stack here, we can visit leaves as we descend left in a single loop.